grocerytracker/GroceryView.swift

```diff
@@ -35,8 +39,20 @@ struct GroceryView: View {
                         }
                     }
                 }
-                .onDelete(perform: removeCategory)
+                .onDelete { indexSet in
+                    deletedIndexSet = indexSet
+                    showingDeleteAlert = true
```

Comment:
hmm, this is the issue, changing 2 state properties. A change to a state prop tells the view to re-render. Somehow this is causing the showingDeleteAlert to get set to false. I wonder what the solution is

Reply:
crazy question, I wonder if reversing these makes a difference

Reply:
StackOverflow says this works, but it is hacky
DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
                 showingDeleteAlert = true
                 }